File: ios/Reguerta/Reguerta/Domain/Access/MemberRepository.swift
import Foundation

protocol MemberRepository: Sendable {
    func member(id: String) async throws -> Member?
    func members(visibleTo member: Member) async throws -> [Member]
    func updateOwnProducerCatalogEnabled(member: Member, enabled: Bool) async throws -> Member
}

protocol LocalMemberRepository: MemberRepository {
    func findByEmailNormalized(_ emailNormalized: String) async -> Member?
    func findByAuthUid(_ authUid: String) async -> Member?
    func linkAuthUid(memberId: String, authUid: String) async -> Member?
    func allMembers() async -> [Member]
    func upsert(member: Member) async -> Member
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/access/MemberRepository.kt
package com.reguerta.user.domain.access

interface MemberRepository {
    suspend fun findByAuthUid(authUid: String): Member?

    suspend fun getMembersVisibleTo(member: Member): List<Member>

    suspend fun updateOwnProducerCatalogEnabled(
        memberId: String,
        isEnabled: Boolean,
    ): Member
}
